app/lib/user_attributes.rb

```diff
@@ -0,0 +1,42 @@
+class UserAttributes
+  CONFIG_KEYS = %w[is_stored_locally].freeze
+
+  attr_reader :attributes
+
+  def initialize
+    @attributes = YAML.safe_load(File.read(Rails.root.join("config/user_attributes.yml"))).with_indifferent_access
+
+    unless Rails.env.production?
+      test_attributes = YAML.safe_load(File.read(Rails.root.join("spec/fixtures/user_attributes.yml"))).with_indifferent_access
+      @attributes.merge!(test_attributes)
+    end
+  end
+
+  def defined?(name)
+    attributes.key? name
+  end
+
+  def stored_locally?(name)
+    attributes.fetch(name)[:is_stored_locally]
+  end
+
+  def errors
+    attributes.each_with_object({}) do |(name, config), errors|
+      config ||= {}
+
+      missing_keys = CONFIG_KEYS.reject { |key| config.keys.include? key }
+      unknown_keys = config.keys.reject { |key| CONFIG_KEYS.include? key }
+      invalid_keys = [
+        [nil, true, false].include?(config["is_stored_locally"]) ? nil : :is_stored_locally,
```

Comment:
nitpick, quite a lot going on here, wonder if it's easier to read if this is off in it's own method?

Reply:
Is afcc7b44069836d3bc8f04cd3e083fc63388ea67 better?